server/src/main/java/org/elasticsearch/tasks/TaskManager.java

```diff
@@ -161,24 +161,32 @@ public Task register(String type, String action, TaskAwareRequest request, boole
             Task previousTask = tasks.put(task.getId(), task);
             assert previousTask == null;
             if (traceRequest) {
-                startTrace(threadContext, task);
+                maybeStartTrace(threadContext, task);
             }
         }
         return task;
     }
 
+    // Start a new trace span if a parent trace context already exists.
+    // For REST actions this will be the case, otherwise Tracer#startTrace can be used.
     // package private for testing
-    void startTrace(ThreadContext threadContext, Task task) {
+    void maybeStartTrace(ThreadContext threadContext, Task task) {
+        if (threadContext.hasParentTraceContext() == false) {
+            return;
+        }
         TaskId parentTask = task.getParentTaskId();
-        Map<String, Object> attributes = Map.of(
-            Tracer.AttributeKeys.TASK_ID,
-            task.getId(),
-            Tracer.AttributeKeys.PARENT_TASK_ID,
-            parentTask.toString()
-        );
+        Map<String, Object> attributes = parentTask.isSet()
+            ? Map.of(Tracer.AttributeKeys.TASK_ID, task.getId(), Tracer.AttributeKeys.PARENT_TASK_ID, parentTask.toString())
```

Comment:
Similarly I like this change but I'm not convinced it should be bundled in the same PR.